File: 0136-single-number/0136-single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        #BEGINEER FRIENDLY DICT
        # count = {}

        # for num in nums:
        #     count[num] = count.get(num, 0) + 1
        
        # for num, freq in count.items():
        #     if freq == 1:
        #         return num
        
        # DEFAULT DICT
        # count = defaultdict(int)
        # for num in nums:
        #     count[num] += 1
        
        # for num,freq in count.items():
        #     if freq == 1:
        #         return num

        #XOR
        result = 0
        for num in nums:
            result ^= num
        
        return result

File: 0136-single-number/0136-single-number.dart
class Solution {
  int singleNumber(List<int> nums) {
    Map<int,int> count = {};

    for (var num in nums)
    {
        count[num] = (count[num] ?? 0) + 1;
    }

    for (var entry in count.entries)
    {
        if(entry.value == 1)
        {
            return entry.key;
        }
    }
    return -1;
    
  }
}